QOpenGLWidget GLSL and WSpinny fixes.

rryan opened the pull request:
Builds on mixxxdj#1863.

Be-ing commented:
GLSL is working again on Linux with Intel graphics, but spinnies are distorted (regardless of scale factor):
![screenshot from 2018-12-29 00-12-32](https://user-images.githubusercontent.com/9455094/50534169-8b817980-0afe-11e9-82aa-393d70ceaff1.png)

rryan commented:
Huh, that's weird. Especially because WSpinny is painted with QPainter, not OpenGL functions.

daschuer commented:
Did you pick the right target branch?
This PR also contains master changes.
I think merging master should be done directly in the original PR.

rryan commented:
> Did you pick the right target branch?
> This PR also contains master changes.
> I think merging master should be done directly in the original PR.

Master has changes that the base branch doesn't have that my branch builds on. I didn't want to wait for @sgielen to merge (and it had some conflicts that I had more context on how to resolve).

rryan commented:
> Did this ever work?
> If not, can we meet this directly to master?

It did -- but with `QOpenGLWidget` it doesn't because `isValid` always returns false until the widget `isVisible`, so I don't think it makes sense in master until @sgielen's PR merges.

rryan commented:
Alternatively I'm happy to take over this work and open a separate PR against mixxxdj/mixxx.

daschuer commented:
>> Did this ever work?
>> If not, can we meet this directly to master?

> It did -- but with QOpenGLWidget it doesn't because isValid always returns false until the widget isVisible, so I don't think it makes sense in master until @sgielen's PR merges.

Do we have an alternative fallback solution?

Holzhaus committed:
CMake: Set more CPack variables

uklotzde committed:
Customize TrackTableBackgroundColorOpacity in <Library> skin node

uklotzde committed:
Upgrade Travis CI Ubuntu images from Xenial to Bionic

uklotzde committed:
Add types for properties of PCM audio signals and streams

uklotzde committed:
Use new PCM audio property types

uklotzde committed:
Get rid of the Janus-headed AudioSignal base class

uklotzde committed:
Defer import of CueInfo until stream properties are known

Holzhaus committed:
mixxx: Finalize MixxxMainWindow on destruction

Holzhaus committed:
util/singleton: Reset m_instance pointer on destruction

This should prevent double-free in case destroy() is called twice.

ninomp committed:
Prevent jump when track analysis finishes while quantization is enabled

daschuer committed:
Added line-length.py to call clang-format for lines > 100 only

daschuer committed:
fix line length to the desired 100 and reduce indentation

daschuer committed:
fix typo

daschuer committed:
use underscore instead of dash

daschuer committed:
Avoid intermediate variable

goddisignz committed:
removed unused and unnecessary methods

nuess0r committed:
Made the dryWetKnob increment/decrement value tweakable

Swiftb0y committed:
added udev rule for AlphaTheta devices

nuess0r committed:
Made it tweakable which VU meter to show Master or Channel. Default is Master

Be-ing committed:
Merge pull request #2609 from Swiftb0y/alpha-theta-udev-rule

added udev rule for AlphaTheta devices

nuess0r committed:
Changed beatLoop encoder to move loop instead of adjust its size

To adjust the size there are the half/double buttons and this behavior is
closer to the mapping of other controllers like the VCI-400

ninomp committed:
Fetch audible start position only when it is needed

ronso0 committed:
restore WPushButton debug tooltip

ronso0 committed:
consolidate WPushButton events

daschuer committed:
Extract threshold parameters and fix to format multible long lines at once.

nuess0r committed:
Added mapping for the CROSSFADER CURVE knob

daschuer committed:
Use capital style with underscores for the Python constants

daschuer committed:
Made the Palette editor a context depending pop up box for the palette combo boxes

daschuer committed:
Seti inital color for QColorDialog

daschuer committed:
Don't translate parentheses

daschuer committed:
Adjust default index when it is out set the new palette selected

daschuer committed:
Merge pull request #2610 from ronso0/pushbutton-tooltip

WPushButton debug tooltip

daschuer committed:
fix typo

Be-ing committed:
ColorPaletteEditor: move Add/Remove Color to buttons

instead of a right click menu where they were not easily
discoverable

Be-ing committed:
ColorPaletteEditor: hide hotcue number column for track palette

It has no meaning in this context.

toszlanyi committed:
cleaned unnecessary loop

goddisignz committed:
Merge branch 'ninomp-lp1855321_additional_fix' into LP1855321_zero_playposition_jump_to_cue

daschuer committed:
Merge pull request #51 from Be-ing/color_preferences

ColorPaletteEditor: move Add/Remove Color to buttons

Holzhaus committed:
scripts/line_length: Use temporary config file

uklotzde committed:
Fix wrong channel count for dual mono layout

uklotzde committed:
Add samples/frames/seconds/milliseconds conversion functions

daschuer committed:
Resort the palette editor buttons. Replace "Save" with "OK"

uklotzde committed:
Improve documentation of basic audio types

daschuer committed:
Shrink add and remove button to + and - and move it below the table.

daschuer committed:
Improve selection behaviour during palette edit

daschuer committed:
Added a warning dialog when removing the palette.

Holzhaus committed:
scripts/line_length.py: Only operate on staged changed lines

Holzhaus committed:
scripts/line_length.py: Pass filename to dump-config

Holzhaus committed:
scripts/line_length: Make sure that paths are correct

Holzhaus committed:
scripts/line_length.py: Pull try-except block out of file ctx manager

uklotzde committed:
Simpliy Travis CI config file

daschuer committed:
Remove commented code

uklotzde committed:
Always finish WebTask with a signal

uklotzde committed:
Add debug output operators for requests and repsonses

uklotzde committed:
Remove implicit auto-delete from web tasks

ronso0 committed:
Merge pull request #2575 from Holzhaus/passthrough-overview

Show "passthrough" on Waveform Overviews if enabled

daschuer committed:
Swap "+" and "-" button

ronso0 committed:
fix powerwindow button release

daschuer committed:
Added some comments

uklotzde committed:
Merge branch 'master' of git@github.com:mixxxdj/mixxx.git into restart_webtask

daschuer committed:
Merge pull request #53 from Holzhaus/line-length-hook

scripts/line_length: Use temporary config file and only operate on staged files

Be-ing committed:
Merge pull request #2614 from ronso0/powerwindow-fix

fix powerwindow button

uklotzde committed:
Don't pass parent to object in a different thread

ronso0 committed:
restore: no play when passthrough is enabled

uklotzde committed:
Cancel all tasks in TagFetcher after last signal

uklotzde committed:
Delete obsolete method deleteAfterFinished()

daschuer committed:
Merge pull request #2534 from uklotzde/restart_webtask

lp1865957: Fetch multiple tracks from MusicBrainz

daschuer committed:
Added Ketan Lambat to the contributor list in the about box. Thank you very much.

uklotzde committed:
Merge branch 'master' into audio_types

uklotzde committed:
Merge branch 'master' into basetracktablemodel

ronso0 committed:
Deere: suppress blue hotcue button border

daschuer committed:
Merge pull request #2620 from ronso0/deere-no-hotcue-border

Deere: suppress blue hotcue button border

daschuer committed:
Merge pull request #52 from Be-ing/color_preferences_hide_hotcue_number

ColorPaletteEditor: hide hotcue number column for track palette

Holzhaus committed:
Merge pull request #2613 from uklotzde/travis_bionic

Upgrade Travis CI Ubuntu images from Xenial to Bionic

daschuer committed:
Use the real hotcue colors when coloring the icon

uklotzde committed:
Merge branch 'master' into basetracktablemodel

uklotzde committed:
Merge branch 'master' into audio_types

Holzhaus committed:
engine/enginedeck: Fix broken vinyl passthrough

In commit 0f5d4b358f023b7431009e13e6b9370794c759ab we connected a
valueChangeRequest that sets the value on the same object that we expect
to send the valueChanged signal. If a value is set for a control object,
that object doesn't emit the valueChanged signal though - it emits the
valueChangedFromEngine signal instead.

uklotzde committed:
Clarify function usage comments

Holzhaus committed:
Merge pull request #2615 from ronso0/no-play-with-passthrough

restore: no play when passthrough is enabled

Holzhaus committed:
scripts/line_length: Fix redefinition of variable

uklotzde committed:
Move debug assertion into base class

uklotzde committed:
Delete invalid debug assertion

toszlanyi committed:
Update controller name

Co-Authored-By: Jan Holthuis <holthuis.jan@googlemail.com>

daschuer committed:
Merge pull request #2622 from Holzhaus/fix-broken-passthrough

engine/enginedeck: Fix broken vinyl passthrough

daschuer committed:
Merge pull request #54 from Holzhaus/line-length-hook

scripts/line_length: Fix redefinition of variable

Holzhaus committed:
Merge pull request #2608 from daschuer/line-length-hook

Added line-length.py to call clang-format for lines > 100 only

Holzhaus committed:
Merge branch '2.2' of github.com:mixxxdj/mixxx

toszlanyi committed:
changelog entry

toszlanyi committed:
Merge branch 'Denon_MC7000_mapping' of github.com:toszlanyi/mixxx into Denon_MC7000_mapping

Holzhaus committed:
Merge pull request #2546 from toszlanyi/Denon_MC7000_mapping

Denon MC7000 mapping

Holzhaus committed:
Merge branch '2.2' of github.com:mixxxdj/mixxx

ronso0 committed:
add font: Open Sans bold

ronso0 committed:
fix cropped Mic/Aux labels in LateNight

daschuer committed:
Merge pull request #2624 from ronso0/open-sans-bold

add font: Open Sans bold

nuess0r committed:
Improvements suggested by Holzhaus

- Changed class name to uppercase (DJC4 instead of djc4)
- Improved controller description

nuess0r committed:
Mapping [Library].MoveLeft and MoveRight to SHIFT + LOAD buttons

nuess0r committed:
Added Stanton DJC.4 to the 2.2.4 CHANGELOG

uklotzde committed:
Add typedefs for value_t

ehendrikd committed:
Rekordbox library feature nulls in strings crash fix (#2627)

Remove nulls in strings

nuess0r committed:
Merge branch '2.2' of https://github.com/mixxxdj/mixxx into Stanton-DJC.4

Holzhaus committed:
Merge pull request #2607 from nuess0r/Stanton-DJC.4

Stanton DJC.4 mapping

Holzhaus committed:
Merge branch '2.2' of github.com:mixxxdj/mixxx

uklotzde committed:
Set properties efficiently by passing a universal reference

uklotzde committed:
Use "left-side const" according to coding style

uklotzde committed:
Merge branch 'master' of git@github.com:mixxxdj/mixxx.git into audio_types

ehendrikd committed:
Fix Windows CI build (#2629)

Be-ing committed:
ColorPaletteEditor: use theme's list-add and list-remove icons

uklotzde committed:
Merge branch 'master' into audio_types

uklotzde committed:
Merge branch 'master' into basetracktablemodel

Be-ing committed:
Merge pull request #2589 from daschuer/color_preferences

Color preferences improvements

Be-ing committed:
ColorPaletteEditor: check if theme has list-add/list-remove icons

Be-ing committed:
update macOS build environment to Qt 5.14.1

uklotzde committed:
Fix wrong debug assertion

Be-ing committed:
ColorPaletteEditor: fix segfault

Holzhaus committed:
Merge pull request #2630 from Be-ing/color_preferences_icons

ColorPaletteEditor: use theme's list-add and list-remove icons

uklotzde committed:
Avoid string conversion for debug formatting

uklotzde committed:
Rename getter and corresponding member

Holzhaus committed:
Merge pull request #2538 from uklotzde/basetracktablemodel

Refactoring: Extract BaseTrackTableModel + BaseCoverArtDelegate

uklotzde committed:
Restore binding of replaygain_peak column

Holzhaus committed:
controllers/controllermanager: Rename sanitizeString and move to anon NS

uklotzde committed:
Initialize Rubberband internals in constructor

uklotzde committed:
Merge branch 'master' of git@github.com:mixxxdj/mixxx.git into audio_types2

Holzhaus committed:
scripts/line_length: Only use files filtered by the pre-commit hook

Holzhaus committed:
pre-commit: Rename line_length to line-length for consistency

Holzhaus committed:
controllers/controllermanager: Fix assertion if no preset is configured

Holzhaus committed:
controllers/controllermanager: Use qDebug() instead of qWarning()

Holzhaus committed:
controllers/dlgprefcontroller: Move shared enumeration code to method

Holzhaus committed:
controllers/dlgprefcontroller: Improve some comments

Holzhaus committed:
controllers/controllermanager: Improve device setup code

Holzhaus committed:
controllers/controllermanager: Remove unused variable

uklotzde committed:
Remove unused template declaration

uklotzde committed:
Add remarks about frame boundary alignment requirement

uklotzde committed:
Replace template integer frame/sample counts with SINT

Be-ing committed:
update hidapi to 0.9.0

The original maintainer of hidapi (signal11) stopped responding
to communication for a long time, so hidapi has been forked. It is
now hosted by libusb at
https://github.com/libusb/hidapi
0.9.0 is the first stable release since the fork.

Holzhaus committed:
controllers/dlgprefcontroller: Add icon to user controller presets

uklotzde committed:
Pre-allocate internal SoundTouch buffers in constructor

Holzhaus committed:
controllers/controllerpresetinfoenumerator: Add constructor for single path

Holzhaus committed:
controllers/dlgprefcontroller: Show Mixxx icon for system presets

Holzhaus committed:
controllers/dlgprefcontroller: Disable checkbox if no preset selected

Holzhaus committed:
controllers/dlgprefcontroller: Rename "..." to "No Preset" and add icon

Holzhaus committed:
controllers/controllermanager: Fix typo in debug message

Holzhaus committed:
Merge branch 'master' of github.com:mixxxdj/mixxx into cleanup-fixes

Holzhaus committed:
mixxx: Only destroy StatsManager singleton if it has been created

Holzhaus committed:
util/singleton: Add some debug assertions

Holzhaus committed:
Merge pull request #2394 from Be-ing/remove_script_tab

DlgPrefController: remove Scripts tab

Holzhaus committed:
Merge branch 'master' of github.com:mixxxdj/mixxx into controller-fixes-part2

Holzhaus committed:
controllers/midi/midimessage: Use desc in  MidiInputMapping == operator

Holzhaus committed:
controllers/midi/midimessage: Fix MidiInputMapping == operator

Holzhaus committed:
controllers/controllermanager: Add "(edited)" suffix to changed presets

Holzhaus committed:
controllers: Add some comments to undocumented classes

Holzhaus committed:
controllers/controllermanager: Remove obsolete importScript method

Holzhaus committed:
controllers: Always load scripts from mapping dir (with system as fallback)

Holzhaus committed:
controllers/dlgprefcontroller: Add "(built-in)" to system scripts

uklotzde committed:
Merge pull request #2623 from Holzhaus/cleanup-fixes

Fix cleanup of Singletons and MixxxMainWindow

Be-ing committed:
Merge pull request #2633 from Be-ing/update_hidapi

update hidapi to 0.9.0

Holzhaus committed:
controllers: Move script file search code into ControllerPreset class

uklotzde committed:
Merge pull request #2625 from ronso0/latenight-mic-label

fix cropped Mic/Aux labels in LateNight

Holzhaus committed:
controllers/controllerpreset: Remove leftover qDebug() statement

Holzhaus committed:
controllers/controllerpresetfilehandler: Use for loop instead of foreach

daschuer committed:
Added Evan Dekker to the contributor list in the about box. Thank you very much.

daschuer committed:
Added Edward Kigwana to the contributor list in the about box, Thank you very much.

daschuer committed:
Added Simon Harst to the contributor list in the about box. Tank you very much.

Be-ing committed:
update macOS build environment to Qt 5.14.2

Holzhaus committed:
controllers: Move controller script file handling to FileHandler classes

Be-ing committed:
hack around macOS segfault on startup

https://bugs.launchpad.net/mixxx/+bug/1871238

uklotzde committed:
Delete error prone and ambiguous legacy sample conversion funcs

uklotzde committed:
Update comment and add warning

Be-ing committed:
Merge pull request #2635 from Be-ing/macos_segfault_hack

hack around macOS segfault on startup

Holzhaus committed:
controllers: Make applyPreset initializeScripts param optional

Holzhaus committed:
controllers/controllermanager: Replace foreach for for loops

Holzhaus committed:
controllers/controllerpresetfilehandler: Improve script file search code

Holzhaus committed:
controllers/controllerpresetfilehandler: Remove wrong debug assertion

Holzhaus committed:
controllers/controllerpresetfilehandler: Improve comments

Holzhaus committed:
controllers/dlgprefcontroller: Remove unused  presetDir code

Holzhaus committed:
controllers/controllerengine: Fix warnings for older Qt5 versions

Holzhaus committed:
controllers: Move savePreset() from Controller into Preset classes

Holzhaus committed:
controllers: Move preset saving from ControllerManger to Preferences

Holzhaus committed:
controllers/dlgprefcontroller: Reload user presets after saving

Holzhaus committed:
controllers/dlgprefcontroller: Remove obsolete method

Holzhaus committed:
controllers/dlgprefcontroller: Fix dirty detection in slotApply()

Holzhaus committed:
controllers/dlgprefcontroller: Reset dialog dirty state on apply

Holzhaus committed:
controllers/dlgprefcontroller: Remove wrong dirty state change

Holzhaus committed:
controllers/dlgprefcontroller: Add acomment

uklotzde committed:
Fix typo

uklotzde committed:
Add const/non-const ptr property accessors

uklotzde committed:
Rename functions and members for importing CueInfos

Holzhaus committed:
controllers/controllermanager: Return QFileInfo in findPresetFile()

Holzhaus committed:
Merge branch 'master' of github.com:mixxxdj/mixxx into controller-fixes-part2

Holzhaus committed:
controllers/dlgprefcontroller: Add edited suffix to filenames

    This allows editing user controller scripts (e.g. downloaded
    from the forum) as well.

Holzhaus committed:
controllers/dlgprefcontroller: Tick Enabled checkbox on preset change

Be-ing committed:
update Windows build environment to Qt 5.14.2

Hopefully this fixes a bug with drag and drop:
https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/drag'n'drop.20issue

Holzhaus committed:
controllers/dlgprefcontroller: Display "No scripts" for scriptless presets

Holzhaus committed:
Merge pull request #2593 from goddisignz/LP1855321_zero_playposition_jump_to_cue

Fix LP1855321 - Prevent jumping to main cue

Holzhaus committed:
controllers: Fix comments and include guards

Holzhaus committed:
appveyor: Switch to Python 3

Holzhaus committed:
CMake: Fix install paths on Windows

Holzhaus committed:
CMake: Set some additional variables for the CPack WIX Generator

Holzhaus committed:
appveyor.yml: Remove NSIS installer generation for now

Holzhaus committed:
Merge pull request #2601 from uklotzde/audio_types

Audio types / stream properties / cue import

Be-ing committed:
Merge pull request #2569 from Holzhaus/controller-fixes-part2

Controllers: Don't needlessly copy presets into user directory

Holzhaus committed:
test/cuecontrol_test: Fix compile error due to missing method

The test was added in #2593 but the Track::setSampleRate method was removed in
didn't help detecting this in this case.

ronso0 committed:
Reloop TM2/4: remove linebreaks from mapping description

ronso0 committed:
Merge pull request #2639 from ronso0/reloopTM24-description

Reloop TM2/4: remove linebreaks from mapping description

Be-ing committed:
Merge pull request #2637 from Holzhaus/fix-ubuntu-appveyor-ci

Switch to pip3 for Ubuntu Appveyor CI build

Be-ing committed:
Merge pull request #2638 from Holzhaus/fix-cuecontrol-test

test/cuecontrol_test: Fix compile error due to missing method

daschuer committed:
Disable -Winvalid-constexpr in case of <experimental/optional> is used which has no constexpr constructor.

Be-ing committed:
Merge pull request #2545 from uklotzde/trackcolorbackgroundopacity

Customize TrackTableBackgroundColorOpacity from skins

daschuer committed:
only disable clang warnings if clang is used

uklotzde committed:
Merge pull request #2641 from daschuer/no-invalid-constexpr

Disable -Winvalid-constexpr

Be-ing committed:
Merge pull request #2631 from Holzhaus/line-length-fix

Fix line-length hook working on all changed files

daschuer committed:
Use ptr#NAME whenever the return value is stored

Be-ing committed:
Merge pull request #2501 from Holzhaus/cpack-installer

Fix CPack WIX Installer for Windows

Be-ing committed:
WaveformRenderMark: fix hotcues getting shifted to the right

Commit f024e7587f9b30a0890cb7e497d55bc3ef137f28
(from GitHub PR #2529) changed hotcue creation so hotcues are
labelled with a null default constructed QString instead of an
empty string by default. This triggered a legacy code path in
WaveformRenderMark::generateMarkImage which is no longer needed.
This commit deletes the legacy code path and renders all cue
images using the same drawing code regardless of whether their
label string is empty.

fixes https://bugs.launchpad.net/mixxx/+bug/1870942

Holzhaus committed:
Add Doxygen config file

Holzhaus committed:
Merge pull request #2648 from Be-ing/hotcue_shift_fix

WaveformRenderMark: fix hotcues getting shifted to the right

uklotzde committed:
Merge pull request #2649 from Holzhaus/doxygen

Add Doxygen config file

uklotzde committed:
.gitignore: Exclude files generated by Doxygen

Holzhaus committed:
Merge pull request #2650 from uklotzde/doxygen_gitignore

.gitignore: Exclude files generated by Doxygen

uklotzde committed:
Merge pull request #2645 from daschuer/no_const_ref

Use ptr#NAME whenever the return value is stored

uklotzde committed:
Fix build with EXTRA_METADATA enabled

uklotzde committed:
Pin Xcode version to 9.4.1 on Travis CI

uklotzde committed:
TrackDAO: Add missing const qualifiers

uklotzde committed:
Update dirty flag when merging imported metadata

uklotzde committed:
Reword documentation

Holzhaus committed:
Merge pull request #2654 from uklotzde/trackdao_const

TrackDAO: Add missing const qualifiers

Holzhaus committed:
Merge pull request #2655 from uklotzde/travis_xcode

Pin Xcode version to 9.4.1 on Travis CI

uklotzde committed:
Merge branch 'master' into extra_metadata

Holzhaus committed:
widget/wcolorpicker: Set button pointer to nullptr after delete

Be-ing committed:
Merge pull request #2657 from Holzhaus/fix-wcolorpicker-button-delete

widget/wcolorpicker: Set button pointer to nullptr after delete

uklotzde committed:
Break long lines

uklotzde committed:
Merge branch 'master' of git@github.com:mixxxdj/mixxx.git into extra_metadata

uklotzde committed:
Restore accidentally deleted lines

daschuer committed:
Merge pull request #2652 from uklotzde/extra_metadata

Fix build with EXTRA_METADATA enabled / Update dirty flag

rryan committed:
Merge pull request #15 from daschuer/qopenglwidget

resolve conflicts, Fix OpenGL status

rryan committed:
Merge branch 'master' into qopenglwidget

rryan committed:
Update CMakeLists.txt to include added/removed files.

rryan committed:
Remove unused header inclusion.

rryan committed:
Use a range-based for loop.

rryan committed:
Rename getAtNextSwap to getPlayPositionAtNextSwap to be more explicit.

rryan committed:
Fix mistakes made when merging.

rryan committed:
No need to guard QWidget::update() with isVisible(), since update does this itself.

rryan committed:
Clarify and update some comments. Remove commented debugging printlines.

rryan committed:
Restore QOpenGLWidget::makeCurrent() calls in constructors.

This has no effect on macOS because the widget's QOpenGLContext is not
created until the widget is shown or resized, but its removal may be
causing segfaults on Linux.

rryan committed:
Move all OpenGL work out of constructors.

With QOpenGLWidget, no work should be done in constructors since our
QOpenGLContext is not valid until the widget is shown.

QAbstractOpenGLFunctions::initializeOpenGLFunctions is cheap (only
does work if it has never been called), so we can call it on every
draw() without a performance penalty.

rryan committed:
Revert "Restore QOpenGLWidget::makeCurrent() calls in constructors."

This reverts commit 53f6e16264c31b5449d4c1241f119ff586d5289f.

rryan committed:
Rework WaveformWidgetFactory initialization so that it has access to the QMainWindow.

This removes the need to use QGuiApplication::focusWindow() or to walk
the list of top level windows to get the QMainWindow. It also hides
some implementation details of WaveformWidgetFactory from the
rest of Mixxx.

JosepMaJAZ committed:
Fix the Windows build with Qt 5.14.2.

rryan committed:
Attempt to fix the AppVeyor build when mkdir fails due to %ENVIRONMENTS_PATH% existing.

rryan committed:
Use setFrameRate to set the configured FPS in WaveformWidgetFactory::setConfig.
